File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/javadoc/comment/direct/Java/Python/s936984238_commented.py
import sys

def main():
    data = list(map(int, sys.stdin.read().split()))
    iter_data = iter(data)
    while True:
        n = next(iter_data)
        if n == 0:
            break
        scores = [next(iter_data) for _ in range(n)]
        scores.sort()
        total = sum(scores[1:-1])
        print(total // (n - 2))

if __name__ == "__main__":
    main()
# End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/javadoc/comment/direct/Java/Go/s936984238_commented.go
package main

import (
	"fmt"
	"sort"
)

func main() {
	for {
		scores, ok := read()
		if !ok {
			break
		}
		solve(scores)
	}
}

func read() ([]int, bool) {
	var n int
	fmt.Scan(&n)
	if n == 0 {
		return nil, false
	}
	scores := make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Scan(&scores[i])
	}
	return scores, true
}

func solve(scores []int) {
	sort.Ints(scores)
	sum := 0
	for i := 1; i < len(scores)-1; i++ {
		sum += scores[i]
	}
	fmt.Println(sum / (len(scores)-2))
}

// End of Code
